heroku/data_source_heroku_app.go

```diff
@@ -0,0 +1,106 @@
+package heroku
+
+import (
+	"context"
+	"github.com/cyberdelia/heroku-go/v3"
+	"github.com/hashicorp/terraform/helper/schema"
+	"time"
+)
+
+func dataSourceHerokuApp() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHerokuAppRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"space": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Default:  nil,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Default:  nil,
+			},
+
+			"stack": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Default: nil,
+			},
+
+			"buildpacks": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+
+			"config_vars": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+				},
+			},
+
+			"git_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"web_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"organization": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Default:  nil,
+			},
+		},
+	}
+}
+
+func dataSourceHerokuAppRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*heroku.Service)
+
+	name := d.Get("name").(string)
+	app, err := client.AppInfo(context.TODO(), name)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(time.Now().UTC().String())
```

Comment:
Interesting. Did you pick this up from how other data sources work?

Reply:
@bernerdschaefer I did - I got it from an AWS data source. acm_certificate, I think.

Reply:
Consider `d.SetId("hd-"+time.Now().UTC().String())`?

Reply:
@jmervine are we concerned that there is some sort of potential negative effect from doing it this way, like collision of IDs? I literally copied this from [the code for aws_acm_certificate](https://github.com/terraform-providers/terraform-provider-aws/blob/7c57e17f40c9d8174ffee34cbd36c77701387c1c/aws/data_source_aws_acm_certificate.go#L146), and I assumed (perhaps incorrectly) that they know what they're doing.

Reply:
Nah, I was just thinking about AWS does it, e.g. instances have `i-<id>`. But I defer to you. It was just a thought.